examples/snake-cli/game.rs

```diff
@@ -0,0 +1,90 @@
+use neat_gru::neural_network::nn::NeuralNetwork;
+
+use crate::{
+    apple::Apple,
+    snake::Snake,
+    utils::{distance_to_apple_x, distance_to_apple_y, distance_to_wall_x, distance_to_wall_y},
+};
+
+#[derive(Debug)]
+pub struct Game {
+    snakes: Vec<Snake>,
+    apple: Apple,
+    scores: Vec<f64>,
+}
+
+impl Game {
+    pub fn get_scores(&self) -> Vec<f64> {
+        self.scores.clone()
+    }
+
+    pub fn new(neural_networks: Vec<NeuralNetwork<f64>>) -> Game {
+        let mut snakes = vec![];
+        for net in &neural_networks {
+            snakes.push(Snake::new(net.clone()));
+        }
+        let mut scores = vec![];
+        neural_networks.iter().for_each(|_| scores.push(0.0));
+        Game {
+            snakes,
+            apple: Apple::generate_apple(),
+            scores,
+        }
+    }
+
+    /// Runs the game and if it finishes returns the game at the end
+    pub fn run_game(&mut self) {
+        while !self.game_over() {
+            self.update();
+        }
+    }
+
+    /// Make the snakes make their decision
+    pub fn make_decision(&mut self) {
+        let mut inputs: [f64; 5] = [0., 0., 0., 0., 0.];
+        let cloned_apple = self.apple.clone();
+        // Let each snake make a decision
+        self.snakes.iter_mut().for_each(|s| {
```

Comment:
You can consider multithreading that in the future with `rayon`

Reply:
That's probably a good idea although right now i want to get it to work with one thread